### PR TITLE
Moving rollingSnapshot form xapi to xo

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

File: src/xo.js [hunk 1]
```diff
@@ -794,6 +794,23 @@ export default class Xo extends EventEmitter {
     return backupFullPath
   }
 
+  async rollingSnapshotVm (vm, tag, depth) {
+    const reg = new RegExp('^rollingSnapshot_[^_]+_' + escapeStringRegexp(tag))
+    const snapshots = sortBy(filter(vm.$snapshots, snapshot => reg.test(snapshot.name_label)), 'name_label')
```
Comment (marsaud) [inline src/xo.js]: I'm not sure what is vm.$snapshots in this context, UUIDs or snapshot objects...

Comment (julien-f) [inline src/xo.js]: Be careful, you are using XO objects as if they were XAPI objects.

You have to get the XAPI object first:

```js
const xapi = this.getXAPI(vm)

// Get the XAPI object for this VM:
vm = xapi.getObject(vm.id)
```

File: src/xo.js [hunk 2]
```diff
@@ -794,6 +794,23 @@ export default class Xo extends EventEmitter {
...
+    const date = safeDateFormat(new Date())
+
+    const vmXapi = this.getXAPI(vm)
+
+    await vmXapi._snapshotVm(vm, `rollingSnapshot_${date}_${tag}_${vm.name_label}`)
```
Comment (julien-f) [inline src/xo.js]: Use the public method: `vmXapi.snapshotVm(vm.$id, ...)`

If the `nameLabel` parameter is missing, feel free to expose it.

File: src/xapi.js [hunk 3]
```diff
@@ -979,35 +976,17 @@ export default class Xapi extends XapiBase {
     }
   }
 
-  async snapshotVm (vmId) {
+  async snapshotVm (vmId, name_label = undefined) {
```
Comment (julien-f) [inline src/xapi.js]: Use camel casing, please :)

File: src/xo.js [hunk 4]
```diff
@@ -794,6 +794,24 @@ export default class Xo extends EventEmitter {
...
 
+  async rollingSnapshotVm (vm, tag, depth) {
+    const xapi = this.getXAPI(vm)
+    vm = xapi.getObject(vm.id)
+
+    const reg = new RegExp('^rollingSnapshot_[^_]+_' + escapeStringRegexp(tag))
+    const snapshots = sortBy(filter(vm.$snapshots, snapshot => reg.test(snapshot.name_label)), 'name_label')
+    const date = safeDateFormat(new Date())
+
+    await xapi.snapshotVm(vm, `rollingSnapshot_${date}_${tag}_${vm.name_label}`)
```
Comment (julien-f) [inline src/xo.js]: `Xapi#snapshotVm()` expects an VM identifier (`vm.$id`), not an object.

**Comment by julien-f** [comment]
LGTM.

**Pull request merged**